app/controllers/comments_controller.rb

```diff
@@ -1,8 +1,7 @@
 class CommentsController < ApplicationController
   before_action :load_comment, only: %i(destroy edit update)
-  before_action :logged_in_user, only: :create
-  include SessionsHelper
-
+  before_action :authenticate_user!, only: :create
+  authorize_resource
```

Comment:
hàm này để làm gì anh

Reply:
anh override lại cái authenticate_user! của gem devise, nếu user đã log in thì chứng thực user, còn nếu chưa log in thì hiển thị ra cái flash báo cần đăng nhập và redirect về trang log in